File: Samples/DMSpeechKitVASampleObjC/DMSpeechKitVASampleObjC/UI/LoginViewController.m
//
//  LoginViewController.m
//  DMSpeechKitVASampleObjC
//
//  Copyright © 2019 Nuance Communications Inc. All rights reserved.
//

#import "LoginViewController.h"
#import "Constants.h"
#import "VAManager.h"
#import "SceneDelegate.h"

@interface LoginViewController ()

@property (weak, nonatomic) IBOutlet UITextField *usernameTextField;
@property (weak, nonatomic) IBOutlet UILabel *errorMessageLabel;

@end

@implementation LoginViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.usernameTextField.text = [[NSUserDefaults standardUserDefaults] valueForKey:USERDEFAULTS_USERNAME_KEY];
}

- (IBAction)loginButtonTapped:(id)sender {
    NSString *username = self.usernameTextField.text;
    if (username.length <= 0) {
        [self.errorMessageLabel setHidden:NO];
        return;
    }
    
    [self storeUsername:username];
    // Initialize session once username is available
    [[VAManager sharedInstance] initializeSession];
    [self showHomeScreen];
}

- (void)storeUsername:(NSString *)username {
    [[NSUserDefaults standardUserDefaults] setValue:username forKey:USERDEFAULTS_USERNAME_KEY];
}

- (void)showHomeScreen {
//    = self.view.window?.windowScene?.delegate as? SceneDelegate
    SceneDelegate *appDelegate = (SceneDelegate *)[[[self.view window] windowScene] delegate];
    [appDelegate loadHomeScreen];
}

@end

File: Samples/DMSpeechKitVASample/DMSpeechKitVASample/UI/ViewControllers/LoginViewController.swift
//
//  LoginViewController.swift
//  DMSpeechKitVASample
//
//  Copyright © 2019 Nuance Communications Inc. All rights reserved.
//

import UIKit

class LoginViewController: UIViewController {

    @IBOutlet weak var usernameTextField: UITextField!
    @IBOutlet weak var errorMessageLabel: UILabel!
    let vaManager = VAManagerImpl.shared
    
    override func viewDidLoad() {
        super.viewDidLoad()
        usernameTextField.text = UserDefaults.standard.string(forKey: Constants.USERDEFAULTS_USERNAME_KEY)
    }
    
    @IBAction func loginButtonTapped(_ sender: Any) {
        guard let username = usernameTextField.text, !username.isEmpty else {
            errorMessageLabel.isHidden = false
            return
        }
        storeUsername(username)
        // Initialize session once username is available
        vaManager.initialize()
        showHomeScreen()
    }
}

extension LoginViewController {
    private func storeUsername(_ username: String) {
        UserDefaults.standard.set(username, forKey: Constants.USERDEFAULTS_USERNAME_KEY)
    }
    
    private func showHomeScreen() {
        if let appDelegate = self.view.window?.windowScene?.delegate as? SceneDelegate {
            appDelegate.loadHomeScreen()
        }
    }
}
